connector/queue/model.py

```diff
@@ -387,3 +403,96 @@ def requeue(self):
+                                string='Parent Channel',
+                                ondelete='restrict')
+    job_function_ids = fields.One2many(comodel_name='queue.job.function',
+                                       inverse_name='channel_id',
+                                       string='Job Functions')
+
+    _sql_constraints = [
+        ('name_uniq',
+         'unique(complete_name)',
+         'Channel complete name must be unique'),
+    ]
+
+    @api.one
+    @api.depends('name', 'parent_id', 'parent_id.name')
+    def _compute_complete_name(self):
+        if not self.name:
+            return  # new record
+        channel = self
+        parts = [channel.name]
+        while channel.parent_id:
+            channel = channel.parent_id
+            parts.append(channel.name)
+        self.complete_name = '.'.join(reversed(parts))
+
+    @api.one
+    @api.constrains('parent_id')
+    def parent_required(self):
+        if self.name != 'root' and not self.parent_id:
+            raise exceptions.ValidationError(_('Parent channel required.'))
+
+    @api.multi
+    def write(self, values):
+        for channel in self:
+            if (not self.env.context.get('install_mode') and
+                    channel.name == 'root' and
+                    ('name' in values or 'parent_id' in values)):
+                raise exceptions.Warning(_('Cannot change the root channel'))
+        return super(JobChannel, self).write(values)
+
+    @api.multi
+    def unlink(self):
+        for channel in self:
+            if channel.name == 'root':
+                raise exceptions.Warning(_('Cannot remove the root channel'))
+        return super(JobChannel, self).unlink()
+
+    @api.multi
+    def name_get(self):
+        result = []
+        for record in self:
+            result.append((record.id, record.complete_name))
+        return result
+
+
+class JobFunction(models.Model):
+    _name = 'queue.job.function'
+    _description = 'Job Functions'
+    _log_access = False
+
+    @api.model
+    def _default_channel(self):
+        return self.env.ref('connector.channel_root')
+
+    name = fields.Char(select=True)
+    channel_id = fields.Many2one(comodel_name='queue.job.channel',
+                                 string='Channel',
+                                 required=True,
+                                 default=_default_channel)
+    channel = fields.Char(related='channel_id.complete_name',
+                          store=True,
+                          readonly=True)
+
+    @api.model
+    def _setup_complete(self):
+        super(JobFunction, self)._setup_complete()
+        for func in JOB_REGISTRY:
+            if not is_module_installed(self.pool, get_openerp_module(func)):
+                continue
+            func_name = '%s.%s' % (func.__module__, func.__name__)
+            if not self.search_count([('name', '=', func_name)]):
+                self.create({'name': func_name})
```

Comment:
@sbidoul here it is https://github.com/acsone/connector/pull/3
I also added tests on the job function and channel models.